torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -379,7 +371,7 @@ void DistEngine::cleanupBackwardPass(const ContextPtr& autogradContext) {
   // not leaking any references to the gradients anywhere else.
   const auto& futureGrads =
       autogradContext->retrieveGraphTask()->future_result_;
-  TORCH_INTERNAL_ASSERT(futureGrads.use_count() == 1);
+  TORCH_INTERNAL_ASSERT(futureGrads.use_count() <= 2);
```

Comment:
No, there is a throw-away test case where this doesn't happen. Part of me wanted to get rid of the check altogether, since asserting the number of references in non-test-case can be brittle.